extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -285,12 +286,18 @@ export class ResourceTypePickerDialog extends DialogBase {
 		this.resourceTypeService.startDeployment(this.getCurrentProvider());
 	}
 
-	private installTools(): void {
+	private async installTools(): Promise<void> {
 		this._dialogObject.message = {
 			level: azdata.window.MessageLevel.Information,
 			text: localize('deploymentDialog.InstallingTools', "Required tools are being installed now.")
 		};
 		this._installToolButton.enabled = false;
 		// TODO
+		for (let i: number = 0; i < this._tools.length; i++) {
+			if (!this._tools[i].isInstalled) {
+				await this._tools[i].install();
```

Comment:
Do these tools need to be installed in serial, or can we do a Promise.all on them?

Reply:
It is be design that I chose to do them serially. Because that guarantees more predictable behavior. We all agree that it is better for us to do them parallelly then I will make the change.